### DISCUSSION TITLE
C-Api capture name from query

### DISCUSSION THREAD

**Discussion by Jack-Punter** [tree-sitter/tree-sitter#1410] [Q&A] [answered]
I may be missing something really obvious here, but i am really struggling to get capture names out of my expressions.

I see there is the `const char *ts_query_capture_name_for_id(const TSQuery *, uint32_t id, uint32_t *length);` however, it is really unclear what id should be, given that the only ID which you get executing a query is the `TSQueryMatch.id`. which is not the case.
The closest i've gotten so far is to use the `TSQueryMatch.pattern_index` as the ID which gives me the first capture name in the expression for that pattern. however, i am yet to figure out how to get the capture name from my match when there are more than one capture per pattern.

```c
ts_query_cursor_exec(query_cursor, query, root);
TSQueryMatch match;
u32 capture_index;
while (ts_query_cursor_next_capture(query_cursor, &match, &capture_index))
{
    TSQueryCapture capture = match.captures[capture_index];
    TSNode node = capture.node;
    u32 length;
    const char *tmp = ts_query_capture_name_for_id(query, match.pattern_index, &length);
    // ...
}
```

**Comment by maxbrunsfeld** [comment, marked as answer]
Yeah, maybe the `id` vs `index` naming is confusing. You use the `TSQueryCapture.index` field to look up the name:

https://github.com/tree-sitter/tree-sitter/blob/2bee7c9b75e3e0163b321502f1f73e2e38943a7e/lib/include/tree_sitter/api.h#L104-L107

**Reply by Jack-Punter** [reply]
🤦 Thank you! I don't know why i didn't think to try using that tbh.
